### PR TITLE
refactor: move not business related code to @masknet/utils

### PR DESCRIPTION
**Pull request by UncleBill**
maintain it as a context less package

## Description

We have a `@masknet/share-base`, but it depends on `@masknet/web3-shared-evm` which is a business related module.
Our codebase really needs a module to hold all context less utility code.

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "@masknet/public-api": "workspace:*",
     "@masknet/sdk": "workspace:*",
     "@masknet/shared": "workspace:*",
+    "@masknet/utils": "workspace:*",
```
Comment (guanbinrui) [inline packages/mask/package.json:31]: Yet another package for utilities.
 
+ @dimensiondev/kit
+ @masknet/shared
+ @masknet/shared-base

Comment (UncleBill) [reply packages/mask/package.json:31]: I update description  to answer this:

> We have a @masknet/share-base, but it depends on @masknet/web3-shared-evm which is a business related module.
Our codebase really needs a module to hold all context less utility code.

Comment (guanbinrui) [reply packages/mask/package.json:31]: Merge utilities from @dimensiondev/kit into this?

File: packages/utils/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@masknet/utils",
+  "description": "This is a context less utils package, it should not contain any business related code.",
```
Comment (Jack-Works) [inline packages/utils/package.json:3]:
```suggestion
  "description": "This is a utility package. It should not have side effect when imported or depends on some environment-specific library like React or @emotion.",
```

File: packages/utils/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Mask utils
+
+This is a context less utils package, it should not contain any business related code. We should make sure it can be use in any other projects.
```
Comment (Jack-Works) [inline packages/utils/README.md:3]:
```suggestion
This is a utility package. It should not have side effect when imported or depends on some environment-specific library like React or @emotion.
```

File: packages/mask/package.json (hunk 1 above)
Comment (Jack-Works) [reply packages/mask/package.json:31]: > We have a @masknet/share-base, but it depends on @masknet/web3-shared-evm which is a business related module.

Actually, I intended to make `@masknet/shared-base` to be context-less.

> @masknet/shared

This package is not for utilities. It's for shared code between dashboard and main extension

**Comment by UncleBill** [comment]
We can merge this after 2.5.0.

File: tsconfig.json [hunk 4]
```diff
@@ -17,7 +17,8 @@
     { "path": "./packages/encryption/tsconfig.tests.json" },
     { "path": "./packages/shared-base/tsconfig.tests.json" },
     // All plugins should be type checked too
-    { "path": "./packages/plugins" }
+    { "path": "./packages/plugins" },
+    { "path": "./packages/utils" }
```
Comment (Jack-Works) [inline tsconfig.json]: Remove this line. It's not an entry project.

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Comment by UncleBill** [comment]
@Jack-Works I notice you moved some utility function to `@dimentiondev/kit`, since `DimensionDev/Maskbook` is the most actively repo, I think we better keep them in this repo. Also agree with @guanbinrui's opinion

>Merge utilities from @dimensiondev/kit into this? Monorepo is more accessible than an external package.

Or, maybe we could share them among multiple repos with `git-subtree` which allows us push changes back to the individual repo.

**Comment by Jack-Works** [comment]
@UncleBill the kit repo is more stable than this one. All utilities in that repo is not likely to change and do not have any business logic. I think it's good to keep that as the status quo.

File: packages/utils/src/worker/index.ts [hunk 5]
```diff
@@ -6,9 +6,9 @@
  */
 export interface OnDemandWorker extends Worker {
```
Comment (Jack-Works) [inline packages/utils/src/worker/index.ts:7]: why you're moving this file? is anywhere else using this feature?

File: packages/utils/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export * from './type'
```
Comment (Jack-Works) [inline packages/utils/src/index.ts:1]: why you're moving type/prototype.ts? that file is hacky and I don't want new code to use them without appropriate reason

**Comment by UncleBill** [comment]
Too old to continue. Will create another one if needed.

**Pull request closed without merging**